### DISCUSSION TITLE
Package per Controllo Carichi

### DISCUSSION THREAD

**Discussion by andbad** [indomus/forum#83] [HUB: Home Assistant]
Qui la vecchia discussione: https://spectrum.chat/indomus/home-assistant/package-per-controllo-carichi~90137ff8-5e3e-4ef2-82db-3e81056786cd

Questo è il link al package: https://github.com/andbad/HA_PowerControl

Per info, suggerimenti o supporto scrivete pure.

By(t)e

**Comment by andbad** [comment]
Inizio io: qualcuno ha avuto modo di provare la "V3" (cioè questa: https://github.com/andbad/HA_PowerControl/tree/v3)?

Oramai è qualche settimana che la testo e sembra funzionare perfettamente. I nuovi limiti temporali lavorano come dovrebbero e non ho mai subito distacchi.

By(t)e

**Reply by bubez81** [reply]
Che splendido lavoro! Testo e faccio sapere, anche se lo vorrei usare un ufficio ed è complicato fare tante prove

**Reply by Marcellino78** [reply]
Lo sto usando anch'io.. vi faro sapere

**Comment by Maspar79** [comment]
Ciao, sono appena passato al nuovo package, farò delle prove in questi giorni, ti farò senz'altro sapere qualcosa, grazie ancora per il tuo prezioso lavoro.

**Reply by Maspar79** [reply]
Ciao andbad funziona egregiamente. Al prossimo package 👍

**Comment by motogm** [comment]
Ciao io in questi gg provero il tuo pacchetto,anche io sono agli inizi (mi faro aiutare),ma per funzionare devo per forza avere un sensore su ogni carico o mi basta quello totale?,mi va bene all’inizio anche se stacca per priorità indifferentemente se è attivo o meno.ciao!

**Reply by motogm** [reply]
ciao no, il suo lo trovo più "limitato" per l'utilizzo che serve a me, mentre il tuo penso sia perfetto. Gestivo 4 teleruttori con il sistema vimar by-me con controllo carichi, ma si è bruciato l'ennesimo modulo e quindi voglio iniziare ad abbandonarla. Io voglio che quando sta iniziando ad arrivare alla soglia stacchi e poi riattacchi, il tuo ha inoltre il vantaggio di staccare solo i carichi che effettivamente stanno consumando, quindi penso sia perfetto. Ora prendo 2 shelly m cosi configuro e ti chiedo

**Reply by andbad** [reply]
Si, ti capisco. Il mio pacchetto però per funzionare ha necessità di sapere se un carico è attivo (quindi sta assorbendo e quanto) oppure no. E per farlo serve un sensore per ogni carico che vuoi monitorare e staccare. Io uso gli Shelly 1PM e Plug S, ma puoi usare quello che vuoi a patto che tu possa derivarne un sensore che legga i W in uso.

By(t)e

**Reply by motogm** [reply]
ciao. Ho preso delle pinze da collegare ai relè che gestiscono lo stacco per forno, lavastoviglie, prese bagno e induzione cucina e un sonoff 4 ch (me lo hanno tasmotato). Per configurare il tutto, siccome io sono abbastanza limitato, puoi darmi un aiuto? ho letto sulla pagina di download del tuo package, dove dai certi passaggi per scontato giustamente, ma io sono a livelli bassi.. Ultima cosa, per caso prima che inizi ad installare, hai in previsione a breve di rilasciare qualche aggiornamento? ciao!

**Reply by andbad** [reply]
Se sei alle prime armi, prima di imbarcarti in questa impresa, è bene che impari a gestire bene Home Assistant, così da capire anche solo i termini o dove mettere i file.
Non che installare questo pacchetto sia complicato, però non è nemmeno la prima cosa che consiglierei di fare ad un neofita. :)

By(t)e

**Comment by andbad** [comment]
Segnalo un piccolo aggiornamento per bugfix: in pratica, in caso si superassa la soglia di distacco immediato, venivano distaccati i carichi fino a rientrare sotto la soglia di distacco ritardato.
Ora invece se si supera la soglia di distacco immediato, vengono scollegati carchi per rientrare sotto quel limite.
Come al solito, potete scaricare il pacchetto aggiornato nel repository del primo post.

By(t)e

**Reply by Marcellino78** [reply]
![Immagine 2021-07-02 235850](https://user-images.githubusercontent.com/58190631/124332350-9155eb00-db91-11eb-8412-49ef6b1abeaf.png)

**Reply by Marcellino78** [reply]
ho questo problema è normale? con l'ultima versione

**Reply by motogm** [reply]
anche io ho lo stesso problema

**Reply by andbad** [reply]
Avete anche ragione... :)
Aggiornate anche il pannello dell'interfaccia (copiate il contenuto del file pc.lovelace nella configurazione del vostro lovelace).

By(t)e

**Reply by Marcellino78** [reply]
niente... io le ho prvate di tutte, eliminato tutto il vecchio.. riavviato, messo il nuovo.... ma niente da fare, rimane l'errore!!!

**Reply by motogm** [reply]
Io invece ho sostituito tutto il file e il problema è sparito.

**Reply by andbad** [reply]
Se ti da sempre lo stesso errore, il problema è nelle schede di lovelace, si vede che non hai copiato bene il codice.
Prova a ricontrollare.

By(t)e

**Comment by andbad** [comment]
Nuova versione:
https://github.com/andbad/HA_PowerControl/releases/tag/v3.0

Changelog:
Ora si possono gestire fino a 20 carichi.
Ottimizzata la logica di distacco
Qualche modifica di dettaglio

**Reply by Maspar79** [reply]
Proverò più tardi, grazie come sempre 😁

**Reply by Maspar79** [reply]
Andbad è sbagliato il link che hai postato, scaricando da questo url: https://github.com/andbad/HA_PowerControl/releases/tag/3.0 rimanda alla vecchia versione. Scaricando invece dal repository del primo post è tutto corretto.

Ciao

**Reply by Maspar79** [reply]
Appena provato sembrerebbe tutto ok. Ciao

**Reply by andbad** [reply]
Grazie della segnalazione, ho modificato il link nel messaggio precedente.

By(t)e

**Reply by Marcellino78** [reply]
Dall'ultimo aggiornamento da errore !!!
Template warning: 'int' got invalid input 'unknown'
Aspetto con pazienza, perche per il resto è una bomba!!

**Reply by andbad** [reply]
Intendi l'ultimo aggiornamento di Home Assistant?

By(t)e

**Comment by Marcellino78** [comment]
Si

Il Mer 3 Nov 2021, 22:50 andbad ***@***.***> ha scritto:

**Reply by andbad** [reply]
Ho aggiornato all'ultima versione anche io e non ho errori, solo warning perché manca il valore di default. Comunque non impatta sul funzionamento (almeno fino alla versione 2022.1, quindi dovrebbe funzionarti.
Comunque lo sistemerò

By(t)e

**Reply by Marcellino78** [reply]
Mi da questo errore... credo dipenda dall'ultimo aggiornamento..

Template warning: 'int' got invalid input 'unknown' when rendering template '{{ states(states('input_text.carico_1_potenza'))|int + states(states('input_text.carico_2_potenza'))|int + states(states('input_text.carico_3_potenza'))|int + states(states('input_text.carico_4_potenza'))|int + states(states('input_text.carico_5_potenza'))|int + states(states('input_text.carico_6_potenza'))|int + states(states('input_text.carico_7_potenza'))|int + states(states('input_text.carico_8_potenza'))|int + states(states('input_text.carico_9_potenza'))|int + states(states('input_text.carico_10_potenza'))|int + states(states('input_text.carico_11_potenza'))|int + states(states('input_text.carico_12_potenza'))|int + states(states('input_text.carico_13_potenza'))|int + states(states('input_text.carico_14_potenza'))|int + states(states('input_text.carico_15_potenza'))|int + states(states('input_text.carico_16_potenza'))|int + states(states('input_text.carico_17_potenza'))|int + states(states('input_text.carico_18_potenza'))|int + states(states('input_text.carico_19_potenza'))|int + states(states('input_text.carico_20_potenza'))|int }}' but no default was specified. Currently 'int' will return '0', however this template will fail to render in Home Assistant core 2022.1

**Reply by andbad** [reply]
Si, esatto. E' un warning, un avviso, non un errore. E ti spiega nell'ultima riga che diventerà errore con la versione 2022.1. Ma per ora funziona.

By(t)e

**Comment by andbad** [comment]
Ecco la versione con i template sistemati (così non vi arrivano più i warning).
https://github.com/andbad/HA_PowerControl/releases/tag/v3.2

By(t)e

**Reply by lucaxxaa** [reply]
complimenti, uso da tempo la vecchia versione. Oggi aggiornero'

**Comment by lucaxxaa** [comment]
Scusate, che vuol dire " Configurare il logger per includere i seguenti sensori:
sensor.potenza_carichi_selezionato
sensor.potenza_carichi_sospesa
sensor.potenza_massima "

Grazie

**Reply by andbad** [reply]
Vuol dire che nella tua configurazione, nella sezione "~~logger~~ recorder" , è consigliato inserire i sensori indicati tra quelli inclusi.
Qui come funziona il componente ~~logger~~ recorder.
~~https://www.home-assistant.io/integrations/logger/~~
https://www.home-assistant.io/integrations/recorder/

Se non hai configurato il ~~logger~~ recorder, di default include tutto, quindi non devi fare altro. Se l'hai configurato, segui il link sopra per includerli.
Non è necessario, il pacchetto funziona lo stesso, ma non avrai i grafici nella pagina di configurazione del pacchetto.
Per intenderci, questo grafico:
![107847400-773a8c80-6deb-11eb-9c08-90e9998ffe08](https://user-images.githubusercontent.com/7837288/141614010-85019214-1c39-4bc6-8887-661803b56e49.png)

EDIT: ah, scusa ho capito, non è il **logger** che devi configurare ma il **recorder**. Errore mio, correggo.

By(t)e

**Reply by lucaxxaa** [reply]
a ecco, si il recorder e correttamente configurato. Il mio e configurato ad esclusione. Mi aevva ingannato logger

Grazie

**Comment by freccia59** [comment]
buongiorno a tutti, sono mauro ed è la prima volta che partecipo e vi ringrazio anticipatamente dell'eventuale aiuto.
ho eseguito tutte le seguenti procedure:
**Copia il file "pc.yaml" nella directory "packages"
Copia il file "update_entities.py" nella directory "python_scripts"
Abilitare i pacchetti
Abilitare gli script python
Aggiungere il contenuto del file "pc.lovelace" all'interfaccia Lovelace.**
Non ho eseguito le successive:
**Creare un gruppo di notifica "notify.tutti" nel file "configuration.yaml" ed inserirvi i dispositivi che riceveranno le notifiche di intervento.
Configurare il recoder per includere i seguenti Sensori:
sensor.potenza_carichi_selezionato
sensor.potenza_carichi_sospesa
sensor.potenza_massima**
Non riesco a capire cosa intenda con:
**Impostare i parametri di configurazione dell'interfaccia grafica Lovelace.**
A questo punto mi compare nella lovelace questa videata:
![image](https://user-images.githubusercontent.com/82373850/146796525-f49a3ac8-f51c-46fa-848c-2353a7acb135.png)
Mentre questa è la schermata di modifica:
![lovelace modifica](https://user-images.githubusercontent.com/82373850/146796966-4b6ae6b7-39f5-47a3-8bcc-bb78e0171bee.jpg)
Non riesco a capire dove sbaglio.
grazie a tutti

**Reply by andbad** [reply]
Due possibilità:
- non hai riavviato HA dopo aver apportato le modifiche
- non hai correttamente abilitato i package

By(t)e

**Reply by freccia59** [reply]
ho riavviato e dimmi se è corretto come ho abilitato il pacchetto
![configyaml](https://user-images.githubusercontent.com/82373850/146818887-39fab673-4d07-4762-bc4d-4f6e8630fe3d.jpg)

**Reply by freccia59** [reply]
Ti ringrazio infinitamente per la cortesia della risposta e la rapidità.
lo sai cosa era. La directory packages aveva la P maiuscola.
Ovviamente le banalità sono quelle che solitamente sfuggono.
grazie ancora
mauro

**Reply by andbad** [reply]
Bene dai, l'importante è aver risolto. :+1: 

By(t)e

**Comment by motogm** [comment]
ciao. Ho un problemino con il tuo controllo carichi. Non so se è avvenuto dopo cheho aggiornato o da quando si era inchiodato HA a causa di codice errato dopo convalida e riavvio che aveva mandato tutto in modalitò provvisoria. Una volta corretto il codice è tutto ripartito. Anche il tuo controllo di carichi, vedo il grafico consumo con variazione andamento, vedo i nuovi carichi aggiuntivi, tutte le voci di regolazione e sono rimaste anche le entità dei sensori di potenza. Non capisco il problema.. Secondo te mi conviene eliminare tutti i file e reistallare il pacchetto? Allego immagine
![Schermata 2021-12-29 alle 10 05 59](https://user-images.githubusercontent.com/83405414/147645824-75ac66cc-a5e5-41df-b43a-92665ebed57c.png)

**Reply by andbad** [reply]
Ciao, scusa per il ritardo (ero in ferie). Lo screen che mi hai mandato è parziale, ma da lì posso vedere che il Sensore Carico Potenza 3 non è configurato, mentre il relativo interruttore si.
Quindi ricontrollerei la configurazione dei vari carichi, come prima cosa.

By(t)e

**Comment by motogm** [comment]
ciao, ora ho resistallato tutto su una nuova versione Hassio per vedere le differenze con core e reistallato il pacchetto e controllato il sensore shelly 3M che non era ben configurato e in sti gg provo. Onestamente non ricordavo più che se c'è un superamento, ma il carico è a 0 il carico forse non viene tagliato. In questi gg rcontrollo. La lavastoviglie non era inserita perchè è staccata in quanto sto cambiando il motore pompa resistenza. Grazie intanto per la risposta.

**Reply by andbad** [reply]
Ti confermo che il carico viene staccato SOLO se viene rilevato che sta assorbendo potenza (più precisamente, se sta assorbendo più di 10W).
Ne consegue che tutti i carichi che vuoi gestire e controllare devono avere il monitoraggio della relativa potenza istantanea.

By(t)e

**Comment by andbad** [comment]
Nuova versione v3.3:
https://github.com/andbad/HA_PowerControl/releases/tag/v3.3

Ho aggiunto un nuovo script python che filtra sensori ed interruttori a seconda del nome. Probabilmente non vi conviene cambiare il nome di tutte le vostre entità, ma non si sa mai (se lasciate l'opzione disabilitata, viene usato il vecchio script).

**Comment by PAOLIG** [comment]
ciao, c'è un anima pia che mi da una mano su come inserire il file pc.lovelace nella mia configurazione? Ho provato a creare una nuova scheda ed incollare il testo ma non riesco... Grazie

**Reply by motogm** [reply]
ciao Paolig. Devi andare sui 3 puntini in alto a dx. cliccare su modifica plancia. DOpo sempre sui 3 puntini in alto clicchi su "editor configurazione testuale" ed inserisci il codice in fondo al resto (avrai altre righe di codice yaml dovuto alla plancia). Attento alla indentazione che sia corretta. Salvi, se non ti diche che ci sono errori, riavvi l'interfaccia e dovrebbe apparirti una nuova plancia dedicata ai carichi

**Reply by PAOLIG** [reply]
Grazie, dopo diverse prove sono riuscito... Il problema erano nelle voci presenti nel file yaml che nell'ultime versione sono un po' cambiate e poi sono diventato pazzo perché lo aggiungevo e come salvavo e mettevo la dashboard non editabile spariva... Mi erano rimaste le righe che portavano i tuoi valori di visibilità. Se posso ti chiedo solo una cosa che non riesco a capire come configurare, il servizio di notifica. Ho aggiunto sotto notify: un notify.tutti, ma non so come configurarlo... ho provato a guardare sulla documentazione di homeassistant ma non riesco a trovare come configurarlo. Grazie in anticipo.

**Reply by PAOLIG** [reply]
Finalmente ce l'ho fatta... scusa ma sono un po' arrugginito. Iniziato 2 anni fa durante il lock down e poi ho fatto qualche aggiunta solo nei periodi che ho tempo, quindi tutte le volte devo iniziare da capo e ricordarmi cosa avevo configurato e poi con tutti questi aggiornamenti sono a fare pulito fra entità che ora hanno integrazione interna e quelle no (ho dovuto rifare automazioni e configurazioni dei package perché dopo 4 aggiornamenti con gli shelly gestiti dal custom shelly for HA, ora gli ho tutti con l'integrazione interna). Grazie comunque e complimenti per il package, veramente ben fatto. Solo un appunto per migliorarlo, se ci fosse la possibilità di "mappare" le entità and un nome scleto sarebbe molto utile al fine di avere gli elettrofdomestici nominati correttamente e non con un nome fuorviante. Per esempio io per comodità gli shelly gli nomino con il macaddress e in nome dove sono utilizzati, ma quando il package gli "chiama", logicamente mi dice distacco 0989080980_shelly_forno al posto del solo nome. Ciao e buon weekend.

**Comment by Symon84** [comment]
Ciao, complimenti per l'ottimo lavoro! Con la versione di HA 2023.3.3 ho notato che se eseguo il riavvio rapido vengono eliminati tutti i sensori e gli switch configurati da interfaccia.
Anche dopo successivo riavvio completo del sistema, il pacchetto non ripristina la configurazione, dovendo pertanto essere riconfigurato manualmente.

Chiedo inoltre se esiste la possibilità di escludere dal riarmo un solo carico (esempio il carico con priorità 1, vorrei che non si riarmasse automaticamente, ma che si spegnesse comunque al superamento della soglia).
E' possibile?
Grazie mille

**Reply by andbad** [reply]
Ciao,
per il primo problema dovrei aver risolto:
https://github.com/andbad/HA_PowerControl/releases/tag/V3.4

Per la seconda richiesta ci lavorerò, ma non restare col fiato sospeso. :)

By(t)e

**Reply by Symon84** [reply]
Ciao, ottimo! 
Ho fatto diverse prove e la configurazione ora, dopo un riavvio rapido non viene del tutto persa.
Dico "non del tutto" in quanto dopo il riavvio rapido i nomi dei sensori vengono tutti sostituiti da un nome generico "Sensore carico priorità X" dove X è il numero del carico...
I nomi dei sensori configurati rimangono nella card "Configurazione salvata", e vengono ripristinati correttamente soltanto dopo un successivo riavvio completo del sistema.
Allego screenshot:
![Schermata 2023-05-18 alle 23 42 35](https://github.com/indomus/forum/assets/97830313/d05fa33e-1fd5-4776-876f-85ad0360ebd3)

Non so se in questo stato ibrido il sistema continui a funzionare, anche se i nomi dei carichi originali non sono riportati..
Mi riservo di fare ulteriori prove di sovraccarico per vedere se comunque lo script "ricorda" la configurazione e vi farò sapere..
Grazie ancora!
Ciao
Simone


EDIT:
mi sono accorto che aggiornare la pagina del browser o dell'app non ha comunque effetto, ma dopo circa 5 minuti i nomi dei sensori vengono ripristinati, quindi non serve un riavvio completo del sistema, basta aspettare un po'!
Concludo confermando che la modifica apportata è efficace ed il bug è stato risolto!
(resta da vedere se durante la fase di aggiornamento, prima di ricompilare i campi dei sensori, lo script risulti comunque operativo, ma se anche fosse, si tratterebbe di una forbice temporale minima e perfettamente tollerabile).
Ottimo lavoro!

**Reply by andbad** [reply]
Ciao, grazie del feedback. :)
Ti confermo che dopo il riavvio c'è un delay di 1 minuto prima che gli script di configurazione agiscano (è necessario per attendere che tutte le entità si aggiornino, altrimenti si rischia di perdere parte della configurazione). Se aggiorni la pagina prima di questo tempo (che si somma al tempo normale di ricaricamento, quindi puoi arrotondare ad un paio di minuti) vedi la configurazione come da tuo screenshot.

Il funzionamento però non ne è comunque influenzato perché la configurazione è presente anche se non te lo mostra nel menu dropdown (come hai notato nella sezione "Configurazione salvata").

By(t)e

**Comment by lucaxxaa** [comment]
E lo uso ormai da tempo il tuo scritto funziona benissimo . Effettivamente con il riavvio rapido cancella tutte le impostazioni.

**Comment by Symon84** [comment]
Ho risolto temporaneamente il problema del riarmo automatico creando un template switch che quando è su off spegne l'entità, mentre quando è su on non esegue nessuna operazione (e torna su off)...
In questo modo posso evitare il riarmo automatico dell'entità, fermo restando che lo script mantiene impegnata la corrente fino a quando attiva il riarmo (anche se non fa nulla), e continua a mandare le notifiche di riarmo (anche se non lo esegue)...
Avevo pensato ad un aiutante booleano da introdurre all'interno dello script python, ma al momento non sono stato in grado di integrarlo per inibire il riarmo, ed evitare la memorizzazione del consumo + notifica di avvenuta riattivazione del carico...
Non resta che "l'accrocchio" dello switch "unidirezionale".. che comunque funziona...

**Comment by andbad** [comment]
Mi hanno segnalato il problema di perdita di configurazione anche [qui ](https://github.com/andbad/HA_PowerControl/issues/27)(non so se è legato al riavvio rapido, ho chiesto).
Devo capire come risolvere, probabilmente "basta" modificare il trigger delle automazioni relative alla configurazione per innescarsi anche in caso di riavvio rapido.
Se avete qualche link che documenti questa nuova funzione sentitevi liberi di segnalarmelo. :)

By(t)e

**Reply by alpitur** [reply]
Ciao, mi spieghi come mettere il trigger per il riavvio rapido? 
Trovo solo homeassistant start or stop

**Reply by Maspar79** [reply]
Ciao, penso che AndBad intendesse una eventuale modifica al suo package.

**Comment by andbad** [comment]
Ho rilasciato una nuova versione che dovrebbe correggere il problema del riavvio rapido:
https://github.com/andbad/HA_PowerControl/releases/tag/V3.4

Fatemi sapere.

By(t)e

**Reply by motogm** [reply]
ciao, funziona sempre il packages? perchè a me pare non andare più. Non so se gli ultimi aggiornamenti di HA hanno corrotto qualcosa

**Reply by andbad** [reply]
Io ho l'ultima versione di HA (2023.12.3) e funziona regolarmente.
Non ho ricevuto altre segnalazioni di problemi particolari.

By(t)e

**Reply by gregpd** [reply]
ciao a tutti, come posso cancellare l'intero pacchetto per reinstallarlo ex novo? grazie

**Comment by Alarico77** [comment]
ciao, sto provando a caricare in una nuova plancia il tuo codice contenuto in pc.lovelace: cancello tutto anche la scritta view:, incollo e quando salvo mi dice: La tua configurazione non è valida: r: At path: views -- Expected an array value, but received: undefined. Cosa sbaglio? grazie.

**Reply by andbad** [reply]
Hai visto questo?
https://github.com/andbad/HA_PowerControl/assets/7837288/73233fa8-2143-4486-bd43-1dce41e59369

By(t)e

**Comment by Alarico77** [comment]
Ecco, dal Mac non si vedeva, da windows si vede. risolto grazie, ne approfitto per farti i complimenti per il lavoro, era proprio quello che cercavo, volevo se posso chiedere come posso fare per impostare la soglia (potenza massimo) che tenga conto della produzione fotovoltaica:

Mi spiego meglio, se imposto 3 kw e casa assorbe 4 kw, lui si attiva, ma se ho ad esempio 1 kw prodotto dal fotovoltaico (o batteria), non vorrei si attivasse, perché la soglia da 3 passerebbe a 4, quindi fino a 4 non si deve attivare.

Forse impostare il (input_number.potenza_massima_immediato), (o ritardato) non con un numero fisso, ma con un valore dinamico si risolve? In questo caso mi creerei un sensore che fa la somma del mio numero fisso, e della produzione.

Non so se dico sciocchezze, casomai perdonami.

Grazie.

Il 04/11/2024 10:56 CET andbad ***@***.***> ha scritto:

**Comment by andbad** [comment]
Dipende da dove leggi la potenza in uso. 
Il caso ottimale sarebbe avere la lettura della potenza prelevata da rete (io lo leggo dalle info fornite dall'inverter), perché è l'esatto valore che ti serve. Se imposti 3 come limite e assorbi 1 da FV, da rete starai assorbendo 2 e quindi PowerControl non entra in azione.

Se invece, per esempio, hai un sensore che ti indica la potenza totale in uso, devi crearti un altro sensore di tipo template dove sottrai alla potenza in uso quella attualmente prelevata da FV o da batteria e dai a PowerControl quest'ultima entità.

By(t)e

**Comment by Alarico77** [comment]
Perfetto grazie ancora, adotterò la seconda soluzione, perchè ho tutti i dati anche dell'inverter, ma ho due regolatori che caricano la batteria, quindi l'inverter non lo sa quanto sto producendo.

Grazie.

**Comment by Peppepalla** [comment]
Ciao, ottimo lavoro........ volevo chiederti se c'e' la possibilita' di inviare le notifiche a dei dispositivi google nest hub, sarebbe ottimo.

**Reply by andbad** [reply]
Un modo sicuramente c'è, ma non avendo google nest hub non saprei aiutarti.

By(t)e

**Comment by alex76m** [comment]
Ciao e complimenti per il progetto, un aiuto se è possibile non riesco a far spegnere i condizionatori perchè non ho uno switch 
i condizionatori sono del BEKO e son integrati con TUYA Local qualcuno può aiutarmi

**Reply by andbad** [reply]
Probabilmente devi creare uno switch di tipo template e poi collegarlo al dispositivo che hai integrato con TUYA.
Purtroppo non sapendo come è integrato mi viene difficile darti indicazioni più precise.

By(t)e

**Comment by DavideVise62** [comment]
Ciao, sto cercando di installarlo dentro la mia istanza di Home Assistant senza però alcun successo.
Ho seguito alla lettera le istruzioni ma non riesco proprio a far funzionare l'integrazione.
Qualcuno saprebbe darmi una mano?

**Reply by KC-inDomus** [reply]
per favore sistema il markup

**Reply by DavideVise62** [reply]
Cosa intendi con markup?

**Reply by andbad** [reply]
Che tipo di problema hai?
Devi dare qualche indicazione più specifica, altrimenti diventa difficile.

By(t)e

**Reply by DavideVise62** [reply]
Certo, scusa.
Allego gli screen della mia config e dell'errore che sto riscontrando.
![Screenshot 2025-03-31 182029](https://github.com/user-attachments/assets/526dfcbd-831b-483f-af63-74360d219e6e)
![Screenshot 2025-03-31 182135](https://github.com/user-attachments/assets/1bfa3b08-46d3-473b-9e45-e79657bc3cea)
![Screenshot 2025-03-31 182144](https://github.com/user-attachments/assets/ea0cd150-f527-434c-843d-e3f0a2cefbdd)

nella scheda del power control non risulta nessuna entità

**Reply by andbad** [reply]
Hai configurato male i packages. Devi avere questo in configuration.yaml
```
homeassistant:
  packages: !include_dir_named packages
```

mentre tu hai
```
homeassistant:
  packages: !include_dir_named packages/pc.yaml
```

By(t)e

**Reply by DavideVise62** [reply]
Grazie per l'aiuto, ho corretto tutto e adesso funziona.
Grazie ancora.

**Reply by DavideVise62** [reply]
Ciao, rispondo nuovamente alla mia segnalazione perché sto ancora riscontrando problemi.
Ho notato che il logbook non funziona più come prima, e mi va a mostrare i log di praticamente nessun sensore.
Altra cosa molto più importante è che il recorder non funziona più come prima e all'interno del package non compare nessun sensore, anche se fino a prima c'erano.
ho già riavviato HA e disattivato il filtro per i sensori.
Seguono gli allegati.
![Screenshot 2025-04-02 224423](https://github.com/user-attachments/assets/0a92734b-4174-41f7-95c9-01f16c5ce398)
![Screenshot 2025-04-02 224403](https://github.com/user-attachments/assets/568a24b7-8138-4c84-b778-411a2b828f2b)

**Reply by andbad** [reply]
Dipende come avevi configurato prima il [recorder](https://www.home-assistant.io/integrations/recorder/).
Questo è come l'hai configurato:
```
recorder:
  include:
    entities:
      - sensor.potenza_carichi_selezionato
      - sensor.potenza_carichi_sospesa
      - sensor.potenza_massima
```
Così hai incluso solo questi sensori (l'ultimo tra l'altro non esiste più, ho aggiornato la [documentazione](https://github.com/andbad/HA_PowerControl/blob/main/README.md)) escludendo tutto il resto.
Se prima non avevi configurato il recorder, probabilmente loggavi tutto. In questo caso ti basta eliminare le righe qua sopra per ripristinare il funzionamento precedente.

------------------------------------------------------------------------------------------
Per quanto riguarda i sensori, verifica se è attiva l'opzione "Nuova modalità selezione opzioni" ed in caso disabilita, salva la configurazione e riavvia Home Assistant.

By(t)e

**Reply by DavideVise62** [reply]
Ciao e scusa per la risposta in ritardo.
Io il recorder non l'ho mai configurato perché preferisco loggare tutto (tanto non ho problemi di archiviazione) e poi eventualmente filtrare in base a quello che mi serve (fasce orarie, entità, ecc.).
Ho rimosso quella parte come mi hai suggerito e i log sono ripresi come prima.
Per quanto riguarda i sensori, non ho mai toccato l'opzione per il filtro, lasciandola sempre disattivata appunto per evitare problemi.
Ho riavviato HA un paio di volte e tutti i sensori sono riapparsi.
Penso sia un problema della mia installazione, con magari qualcosa che si avvia e va in conflitto, in quanto ho riscontrato che riavviando HA, senza cambiare niente, appaio e scompaiono.

**Reply by andbad** [reply]
Se ti va, potresti abilitare i [log](https://github.com/andbad/HA_PowerControl#debug) e mandarmeli (magari apri una [issue](https://github.com/andbad/HA_PowerControl/issues) così non inquiniamo questo thread) così vediamo che succede.

By(t)e

**Reply by DavideVise62** [reply]
Ciao, li abilito e ti tengo aggiornato se il problema dovesse verificarsi nuovamente.

**Comment by alex76m** [comment]
Che altre info posso darti sono integrati tramite tuya

Il lun 31 mar 2025, 17:01 Davis. ***@***.***> ha scritto:

**Reply by andbad** [reply]
Ti ho risposto qui:
https://github.com/andbad/HA_PowerControl/issues/57

By(t)e

**Comment by laurafabry** [comment]
Ciao a tutti ho un problema con il package controllo carichi chiamato Power-control.
mi da questo errore :
 Action script.powercontrol_configurazione_popola_entita not found

cosa posso fare per risolvere? io ho seguito la guida per installare ma HA mi restituisce questo errore

**Reply by andbad** [reply]
[Abilita i log](https://github.com/andbad/HA_PowerControl?tab=readme-ov-file#debug) e postali qui, così vediamo cosa si inceppa.
Indicami anche la versione di Home Assistant che usi e la versione di Power Control che vuoi installare (anche se immagino sia l'ultima).

By(t)e

**Comment by laurafabry** [comment]
questo e' l'errore che mi da:
Registratore: homeassistant.components.automation.powercontrol_configurazione_carica_2
Fonte: helpers/script.py:2057
Integrazione: Automazione ([documentazione](https://www.home-assistant.io/integrations/automation), [problemi](https://github.com/home-assistant/core/issues?q=is%3Aissue+is%3Aopen+label%3A%22integration%3A+automation%22))
Prima occorrenza: 17:46:09 (2 occorrenze)
Ultimo accesso: 17:46:09

PowerControl - Configurazione - Carica: Choose at step 2: choice 1: Error executing script. Service not found for call_service at pos 2: Action script.powercontrol_configurazione_popola_entita not found
PowerControl - Configurazione - Carica: Error executing script. Service not found for choose at pos 2: Action script.powercontrol_configurazione_popola_entita not found

e nella sezione dove selezioni i vari carichi non vedo nulla

**Reply by andbad** [reply]
Come dicevo sarebbe meglio avere i log completi, ma ho il sospetto che tu non abbia installato correttamente il package. Elimina i relativi file riavvia Home assistant, reinstallata il package copiando i file nelle corrette cartelle e modificando il file di configurazione, ed infine riavvia home assistant.

By(t)e

**Comment by andbad** [comment]
Ho migrato il package yaml ad una integrazione HACS.
Sempre al solito link.
https://github.com/andbad/HA_PowerControl

Fatemi sapere.

By(t)e

**Comment by Alarico77** [comment]
hai le istruzioni? non capisco come si imposta tutto

**Reply by andbad** [reply]
Hai guardato il README?

By(t)e
